src/Microsoft.Health.Fhir.Shared.Api/Controllers/ValidateController.cs

```diff
@@ -39,23 +42,58 @@ public ValidateController(IMediator mediator)
         [HttpPost]
         [Route(KnownRoutes.ValidateResourceType)]
         [AuditEventType(AuditEventSubType.Read)]
-        public async Task<IActionResult> Validate([FromBody] Resource resource)
+        public async Task<IActionResult> Validate([FromBody] Resource resource, [FromQuery(Name = "profile")] Uri profile)
         {
-            return await RunValidationAsync(resource);
+            if (resource.ResourceType == ResourceType.Parameters)
+            {
+                var parameterResource = (Parameters)resource;
+                var profileFromParameters = parameterResource.Parameter.Find(param => param.Name.Equals("profile", StringComparison.OrdinalIgnoreCase));
+                if (profileFromParameters != null)
+                {
+                    if (profile != null)
+                    {
+                        throw new BadRequestException(Api.Resources.MultipleProfilesProvided);
+                    }
+
+                    if (profileFromParameters.Value == null)
+                    {
+                        throw new BadRequestException("Profile parameter is empty");
```

Comment:
>"Profile parameter is empty" [](start = 54, length = 28)

Since these messages are returned to the user, they should be included in the `Resources` file (so that they get translated). Here and below.

Reply:
There is also few messages I can remove from Resource class, thanks for reminder!